fw/apm.c

```diff
@@ -643,12 +628,27 @@ typedef struct {
 #define TFW_APM_DFLT_REQ	"\"GET / HTTTP/1.0\r\n\r\n\""
 #define TFW_APM_DFLT_URL	"\"/\""
 
+/*
+ * APM Data structure.
+ *
+ * Note that the organization of the supporting data heavily depends
+ * on the fact that there's only one party that does the calculation
+ * of percentiles - the function that runs periodically on timer.
+ * If there are several different parties that do the calculation,
+ * then the data may need to be organized differently.
+ *
+ * @rbuf	- The ring buffer for the specified time window.
+ * @rbctl	- The control data helpful in taking optimizations.
+ * @stats	- The latest percentiles.
+ * @ubuf	- The buffer that holds data for updates, per CPU.
+ * @timer	- The periodic timer handle.
+ * @flags	- The atomic flags (see below).
```

Comment:
Looks like the flags are no longer below after the rearrangement of the code.

Reply:
Fixed